bundles/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/internal/handler/EvohomeHeatingZoneHandler.java

```diff
@@ -82,16 +89,115 @@ public void handleCommand(ChannelUID channelUID, Command command) {
+                    // Get a local copy of schedule when needed, saves on api calls
+                    DailySchedules schedule = getDailySchedules();
+                    double expectedTemp = getCurrentSetPoint(schedule);
+                    if (newTemp == CANCEL_SET_POINT_OVERRIDE || newTemp == expectedTemp) {
                         bridge.cancelSetPointOverride(getEvohomeThingConfig().id);
+                        updateState(EvohomeBindingConstants.ZONE_SET_POINT_CHANNEL, new DecimalType(expectedTemp));
+                        return;
                     } else if (newTemp < 5) {
                         newTemp = 5;
                     }
                     if (newTemp >= 5 && newTemp <= 35) {
-                        bridge.setPermanentSetPoint(getEvohomeThingConfig().id, newTemp);
+                        int overrideMode = getOverrideMode();
+                        switch (overrideMode) {
+                            case EvohomeBindingConstants.SETPOINT_OVERRIDE_TEMPORARY_SCHEDULE_NEXT:
+                                bridge.setTemporarySetPoint(getEvohomeThingConfig().id, newTemp,
+                                        getNextSetPointChange(schedule));
+                                break;
+                            case EvohomeBindingConstants.SETPOINT_OVERRIDE_TEMPORARY_TIME_ADD:
+                                bridge.setTemporarySetPoint(getEvohomeThingConfig().id, newTemp,
+                                        LocalDateTime.now().plusMinutes(getOverrideTime()));
+                                break;
+                            default:
+                                bridge.setPermanentSetPoint(getEvohomeThingConfig().id, newTemp);
+                                break;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private DailySchedules getDailySchedules() {
+        return this.getEvohomeBridge().getZoneSchedule("temperatureZone", this.getId());
+    }
+
+    private LocalDateTime getNextSetPointChange() {
+        return getNextSetPointChange(this.getDailySchedules());
+    }
+
+    private LocalDateTime getNextSetPointChange(DailySchedules schedule) {
+        if (schedule != null) {
+            int currentWeekDay = LocalDateTime.now().getDayOfWeek().getValue() - 1;
+            LocalDateTime nextSetPointChange = null;
+            for (DailySchedule daily : schedule.getSchedules()) {
+                // handle wrap around
+                int dayDelta = 0;
+                if (daily.getWeekday() < currentWeekDay) {
+                    // the schedule day is from earlier in the week so add 7 to get to the same day next week
+                    dayDelta = daily.getWeekday() + 7 - currentWeekDay;
+                } else {
+                    dayDelta = daily.getWeekday() - currentWeekDay;
+                }
+                LocalDate setPointDay = LocalDate.now().plusDays(dayDelta);
```

Comment:
This could be a method on its own.

Reply:
the code is only used here, the later check is looking for the setpoint BEFORE now, so the comparrison is backwards